docs/api_compatibility.md

```diff
@@ -0,0 +1,47 @@
+# API compatibility
+
+JAX is constantly evolving, and we are not yet ready to commit to a stable API.
+That said, we want to minimize churn for the JAX user community.
+
+JAX follows a 3 month deprecation policy. When an incompatible change is made
+to an API, we will make our best effort to obey the following procedure:
+* the change will be announced in `CHANGELOG.md`, and the old API will issue a
+  `DeprecationWarning`.
+* three months after the `jax` release that deprecated an API, we may remove the
+  deprecated API at any time.
+
+We reserve the right to change this policy at any time.
+
+## What is covered?
+
+Only public JAX APIs are covered, which includes the following modules:
+
+* `jax`
+* `jax.dlpack`
+* `jax.image`
+* `jax.lax`
+* `jax.nn`
+* `jax.numpy`
+* `jax.ops`
+* `jax.profiler`
+* `jax.random`
+* `jax.scipy`
+* `jax.tree_util`
+
+Not everything in these modules is public. Over time, we are working to separate
+public and private APIs.
```

Comment:
Mention that our aim is to follow the convention of hiding all non-public apis behind names that begin with an underscore (although we're not entirely compliant with that yet)

Reply:
I would also note that public APIs are documented in the JAX API docs

Reply:
Both done.